sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -107,6 +107,11 @@ internal static class UnitTestHelper
         public static bool HasExpectedExceptionAttribute(this IMethodSymbol method) =>
             method.GetAttributes().Any(a => a.AttributeClass.IsAny(KnownExpectedExceptionAttributes));
 
+        public static bool HasAssertionInAttribute(this IMethodSymbol method) =>
+            method.GetAttributes().Any(a =>
+                a.AttributeClass.Is(KnownType.NUnit_Framework_TestCaseAttribute) &&
+                a.NamedArguments.Any(arg => arg.Key == "ExpectedResult"));
+
```

Comment:
We can extract a small method here to improve a bit readability:
```suggestion
public static bool HasAssertionInAttribute(this IMethodSymbol method) =>
    method.GetAttributes().Any(IsTestCaseAttributeWithExpectedResult);

private static bool IsTestCaseAttributeWithExpectedResult(AttributeData a) =>
    a.AttributeClass.Is(KnownType.NUnit_Framework_TestCaseAttribute) &&
    a.NamedArguments.Any(arg => arg.Key == "ExpectedResult");
```